pkgs/development/libraries/raylib/default.nix

```
@@ -0,0 +1,39 @@
+{ stdenv, lib, fetchFromGitHub,
+  cmake,
+  alsaLib, mesa, libGLU, glfw,
+  libX11, libXi, libXcursor, libXrandr, libXinerama,
+  includeEverything ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "raylib";
+  version = "3.0.0";
+
+  src = fetchFromGitHub {
+    owner = "raysan5";
+    repo = pname;
+    rev = version;
+    sha256 = "1chj7sril4l2dxh7flp5ndddydbbf1fhnqlydaysm4m6waxidxmr";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ alsaLib mesa libGLU glfw libX11 libXi libXcursor libXrandr libXinerama ];
+
+  # https://github.com/raysan5/raylib/wiki/CMake-Build-Options
+  cmakeFlags = [
+    "-DUSE_EXTERNAL_GLFW=ON"
+    "-DSHARED=ON"
+  ] ++ lib.optional includeEverything "-DINCLUDE_EVERYTHING=ON";
+
+  # disable -Werror=format-security until new version (>3.0.0) is released with this fix
+  # https://github.com/raysan5/raylib/pull/1305
+  hardeningDisable = [ "format" ];
+
```

Should fix the audio problem:
```suggestion
  postFixup = ''
    wrapProgram $out/lib/*.so --prefix LD_LIBRARY_PATH : {alsaLib}/lib
  '';
```

---

Ah, I was looking at this but wasn't sure if it worked for shared objects or just executables.
Fantastic, I'll try this out tomorrow!

---

Ok, so I can get this to work when patching the example binaries, but not the .so files themselves.
```
  postFixup = ''
    wrapProgram $out/../source/build/examples/audio_music_stream --prefix LD_LIBRARY_PATH : ${alsaLib}/lib
  '';
```

If I patch the .so files I get:
```
$ ./audio_music_stream 
./audio_music_stream: error while loading shared libraries: <snip>/out/lib/libraylib.so.301: invalid ELF header
```

---

`wrapProgram` only wraps executables afaik, not dynamic libraries.

> If I patch the .so files I get:
> `[…] invalid ELF header`

If you `cat` that "library", it most likely placed a script there that sets `LD_LIBRARY_PATH` then tries to `exec` the wrapped library. That'll definitely not work for this.